src/v/cluster/partition_balancer_state.h

```diff
@@ -11,6 +11,7 @@
 #pragma once
 
 #include "cluster/fwd.h"
+#include "cluster/node_status_table.h"
```

Comment:
as a matter of reducing header dependencies, I think forward declaration from `cluster/fwd.h` should be enough here. Those .cc files that really use `node_status_table` should include it themselves directly.